File: 1.0/src/c/cz/mg/vulkan/PFN_vkEnumeratePhysicalDeviceGroupsKHX.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkEnumeratePhysicalDeviceGroupsKHX_call(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong pPhysicalDeviceGroupCount, jlong pPhysicalDeviceGroupProperties, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkEnumeratePhysicalDeviceGroupsKHX f = (PFN_vkEnumeratePhysicalDeviceGroupsKHX)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkInstance*)jniLongToPointer(instance)),
        ((uint32_t*)jniLongToPointer(pPhysicalDeviceGroupCount)),
        ((VkPhysicalDeviceGroupPropertiesKHX*)jniLongToPointer(pPhysicalDeviceGroupProperties))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkEnumeratePhysicalDeviceGroupsKHX.java
package cz.mg.vulkan;

public class PFNvkEnumeratePhysicalDeviceGroupsKHX extends VkFunctionPointer {
    public PFNvkEnumeratePhysicalDeviceGroupsKHX() {
    }

    public PFNvkEnumeratePhysicalDeviceGroupsKHX(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkEnumeratePhysicalDeviceGroupsKHX(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkEnumeratePhysicalDeviceGroupsKHX(long value) {
        setValue(value);
    }

    public PFNvkEnumeratePhysicalDeviceGroupsKHX(VkInstance instance) {
        super(instance, new VkString("vkEnumeratePhysicalDeviceGroupsKHX"));
    }

    public void call(VkInstance instance, VkUInt32 pPhysicalDeviceGroupCount, VkPhysicalDeviceGroupPropertiesKHX pPhysicalDeviceGroupProperties, VkResult rval){
        call(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), pPhysicalDeviceGroupCount != null ? pPhysicalDeviceGroupCount.getVkAddress() : VkPointer.NULL, pPhysicalDeviceGroupProperties != null ? pPhysicalDeviceGroupProperties.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getSinkAddressNative());
    }

    protected static native void call(long vkaddress, long instance, long pPhysicalDeviceGroupCount, long pPhysicalDeviceGroupProperties, long rval);
}
